src/ruby/puppetserver-lib/puppet/server/puppet_config.rb

```diff
@@ -22,6 +22,12 @@ def self.initialize_puppet(puppet_config)
     )
     Puppet[:trace] = true
 
+    # (PUP-8141) Without this setting, puppetserver will call out to facter,
+    # which will in turn load all facts, which will in turn fail horribly on
+    # osx looking for cfpropertylist, which won't exist. This is only true
+    # when running from source on osx, for development and tests.
+    Puppet[:digest_algorithm] = "md5"
```

Comment:
Running the server test suite I also had to set `supported_checksum_types`.

Reply:
which test suite? i didn't see any problems running the clojure tests or the specs

Reply:
Oh, the clojure tests fail for me with this PR
```
sellout:ps justin$ git checkout -b mergehaus                                                                                                                                                     [2131/9526]
Switched to a new branch 'mergehaus'                                                                                                                                                                        
sellout:ps justin$ git merge haus/ticket/master/pup-8141-set-digest-algorithm                                                                                              
Updating c3cffaf..cbe56d6                                                                                                                                                                                   
Fast-forward                                                                                                                                                             
 src/ruby/puppetserver-lib/puppet/server/puppet_config.rb | 8 ++++++++                                                                                                     
 1 file changed, 8 insertions(+)                                                                                                                                           
sellout:ps justin$ lein test                                                                                                                                                                           
Using 1G min heap since not set via PUPPETSERVER_HEAP_SIZE environment variable or user.puppetserver-heap-size in ~/.lein/profiles.clj file. Set to at least 5G for best performance during test runs.      
Using 2G max heap since not set via PUPPETSERVER_HEAP_SIZE environment variable or user.puppetserver-heap-size in ~/.lein/profiles.clj file. Set to at least 5G for best performance during test runs.   
Compiling ClojureScript...                                                                                                                                                                                 
Compiling ClojureScript...                                                                                                                                                 
                                                                                                                                                                           
lein test puppetlabs.general-puppet.general-puppet-int-test                                                                                                              
2018-01-11 11:13:34,644 WARN  [clojure-agent-send-pool-0] [puppetserver] Puppet Support for ruby version 1.9.3 is deprecated and will be removed in a future release. See https://docs.puppet.com/puppet/lat
est/system_requirements.html#ruby for a list of supported ruby versions.                                                                                                   
   (at /Users/justin/Backup/code/work/server/ps/ruby/puppet/lib/puppet.rb:130:in `Puppet')                                                                                                                  
2018-01-11 11:13:34,695 WARN  [clojure-agent-send-pool-0] [puppetserver] Puppet Setting app_management is deprecated.                                                      
   (at /Users/justin/Backup/code/work/server/ps/ruby/puppet/lib/puppet/settings.rb:1161:in `issue_deprecation_warning')                                                                                     
2018-01-11 11:13:35,636 WARN  [async-dispatch-2] [o.e.j.s.h.ContextHandler] Empty contextPath                                                                              
2018-01-11 11:13:36,905 WARN  [qtp1685379121-27] [puppetserver] Facter Error loading fact /Users/justin/Backup/code/work/server/ps/ruby/facter/lib/facter/ec2.rb: no such file to load -- cfpropertylist    org/jruby/RubyKernel.java:1040:in `require'                                                                                                                                
/Users/justin/.m2/repository/org/jruby/jruby-stdlib/1.7.27/jruby-stdlib-1.7.27.jar!/META-INF/jruby.home/lib/ruby/shared/rubygems/core_ext/kernel_require.rb:54:in `require'                                 
/Users/justin/Backup/code/work/server/ps/ruby/facter/lib/facter/util/macosx.rb:9:in `Macosx'                                                    
/Users/justin/Backup/code/work/server/ps/ruby/facter/lib/facter/util/macosx.rb:8:in `(root)'                                                         
org/jruby/RubyKernel.java:1040:in `require'                                                                                        
/Users/justin/.m2/repository/org/jruby/jruby-stdlib/1.7.27/jruby-stdlib-1.7.27.jar!/META-INF/jruby.home/lib/ruby/shared/rubygems/core_ext/kernel_require.rb:1:in `(root)'
/Users/justin/.m2/repository/org/jruby/jruby-stdlib/1.7.27/jruby-stdlib-1.7.27.jar!/META-INF/jruby.home/lib/ruby/shared/rubygems/core_ext/kernel_require.rb:54:in `require'
org/jruby/RubyProc.java:281:in `call'
/Users/justin/Backup/code/work/server/ps/ruby/facter/lib/facter/virtual.rb:46:in `(root)'
/Users/justin/Backup/code/work/server/ps/ruby/facter/lib/facter/util/resolution.rb:157:in `resolve_value'
org/jruby/ext/timeout/Timeout.java:98:in `timeout'
/Users/justin/Backup/code/work/server/ps/ruby/facter/lib/facter/core/resolvable.rb:63:in `value'
/Users/justin/Backup/code/work/server/ps/ruby/facter/lib/facter/core/resolvable.rb:62:in `value'
/Users/justin/Backup/code/work/server/ps/ruby/facter/lib/facter/core/resolvable.rb:84:in `with_timing'
/Users/justin/Backup/code/work/server/ps/ruby/facter/lib/facter/core/resolvable.rb:61:in `value'
org/jruby/RubyArray.java:1613:in `each'
/Users/justin/Backup/code/work/server/ps/ruby/facter/lib/facter/util/fact.rb:161:in `find_first_real_value'
/Users/justin/Backup/code/work/server/ps/ruby/facter/lib/facter/util/fact.rb:160:in `find_first_real_value'
/Users/justin/Backup/code/work/server/ps/ruby/facter/lib/facter/util/fact.rb:113:in `value'
/Users/justin/Backup/code/work/server/ps/ruby/facter/lib/facter/util/fact.rb:145:in `searching'
/Users/justin/Backup/code/work/server/ps/ruby/facter/lib/facter/util/fact.rb:110:in `value'
/Users/justin/Backup/code/work/server/ps/ruby/facter/lib/facter/util/collection.rb:130:in `value'
/Users/justin/Backup/code/work/server/ps/ruby/facter/lib/facter.rb:117:in `value'
org/jruby/RubyProc.java:281:in `call'
/Users/justin/Backup/code/work/server/ps/ruby/facter/lib/facter/ec2.rb:23:in `(root)'
/Users/justin/Backup/code/work/server/ps/ruby/facter/lib/facter/util/confine.rb:40:in `true?'
org/jruby/RubyEnumerable.java:1409:in `all?'
/Users/justin/Backup/code/work/server/ps/ruby/facter/lib/facter/core/suitable.rb:111:in `suitable?'
/Users/justin/Backup/code/work/server/ps/ruby/facter/lib/facter/core/suitable.rb:111:in `suitable?'
org/jruby/RubyArray.java:1613:in `each'
org/jruby/RubyEnumerable.java:702:in `find_all'
/Users/justin/Backup/code/work/server/ps/ruby/facter/lib/facter/util/fact.rb:152:in `find_suitable_resolutions'
/Users/justin/Backup/code/work/server/ps/ruby/facter/lib/facter/util/fact.rb:152:in `find_suitable_resolutions'
/Users/justin/Backup/code/work/server/ps/ruby/facter/lib/facter/util/fact.rb:112:in `value'
/Users/justin/Backup/code/work/server/ps/ruby/facter/lib/facter/util/fact.rb:145:in `searching'
/Users/justin/Backup/code/work/server/ps/ruby/facter/lib/facter/util/fact.rb:110:in `value'
/Users/justin/Backup/code/work/server/ps/ruby/facter/lib/facter/util/collection.rb:130:in `value'
/Users/justin/Backup/code/work/server/ps/ruby/facter/lib/facter.rb:117:in `value'
org/jruby/RubyKernel.java:1059:in `load'
/Users/justin/Backup/code/work/server/ps/ruby/facter/lib/facter/ec2.rb:63:in `(root)'
/Users/justin/Backup/code/work/server/ps/ruby/facter/lib/facter/util/loader.rb:1:in `(root)'
/Users/justin/Backup/code/work/server/ps/ruby/facter/lib/facter/util/loader.rb:130:in `kernel_load'
org/jruby/RubyArray.java:1613:in `each'
/Users/justin/Backup/code/work/server/ps/ruby/facter/lib/facter/util/loader.rb:115:in `load_file'
org/jruby/RubyArray.java:1613:in `each'
/Users/justin/Backup/code/work/server/ps/ruby/facter/lib/facter/util/loader.rb:49:in `load_all'
/Users/justin/Backup/code/work/server/ps/ruby/facter/lib/facter/util/loader.rb:47:in `load_all'
/Users/justin/Backup/code/work/server/ps/ruby/facter/lib/facter/util/loader.rb:45:in `load_all'
/Users/justin/Backup/code/work/server/ps/ruby/facter/lib/facter/util/collection.rb:76:in `fact'
/Users/justin/Backup/code/work/server/ps/ruby/facter/lib/facter/util/collection.rb:129:in `value'
/Users/justin/Backup/code/work/server/ps/ruby/facter/lib/facter.rb:117:in `value'
org/jruby/RubyProc.java:281:in `call'
/Users/justin/Backup/code/work/server/ps/ruby/puppet/lib/puppet/defaults.rb:20:in `default_checksum_types'
/Users/justin/Backup/code/work/server/ps/ruby/puppet/lib/puppet/defaults.rb:941:in `Puppet'
/Users/justin/Backup/code/work/server/ps/ruby/puppet/lib/puppet/settings/base_setting.rb:100:in `default'
/Users/justin/Backup/code/work/server/ps/ruby/puppet/lib/puppet/settings.rb:1281:in `lookup'
/Users/justin/Backup/code/work/server/ps/ruby/puppet/lib/puppet/settings.rb:1296:in `interpolate'
org/jruby/RubyHash.java:1403:in `each_key'
/Users/justin/Backup/code/work/server/ps/ruby/puppet/lib/puppet/settings.rb:1082:in `value_sym'
/Users/justin/Backup/code/work/server/ps/ruby/puppet/lib/puppet/parser/scope.rb:722:in `merge_settings'
/Users/justin/.m2/repository/org/jruby/jruby-stdlib/1.7.27/jruby-stdlib-1.7.27.jar!/META-INF/jruby.home/lib/ruby/1.9/forwardable.rb:201:in `each_key'
/Users/justin/Backup/code/work/server/ps/ruby/puppet/lib/puppet/parser/scope.rb:719:in `merge_settings'
/Users/justin/Backup/code/work/server/ps/ruby/puppet/lib/puppet/parser/compiler.rb:781:in `create_settings_scope'
/Users/justin/Backup/code/work/server/ps/ruby/puppet/lib/puppet/parser/compiler.rb:161:in `compile'
/Users/justin/Backup/code/work/server/ps/ruby/puppet/lib/puppet/util/profiler/around_profiler.rb:58:in `profile'
/Users/justin/Backup/code/work/server/ps/ruby/puppet/lib/puppet/util/profiler.rb:51:in `profile'
/Users/justin/Backup/code/work/server/ps/ruby/puppet/lib/puppet/parser/compiler.rb:161:in `compile'
/Users/justin/Backup/code/work/server/ps/ruby/puppet/lib/puppet/context.rb:65:in `override'
/Users/justin/Backup/code/work/server/ps/ruby/puppet/lib/puppet.rb:260:in `override'
/Users/justin/Backup/code/work/server/ps/ruby/puppet/lib/puppet/parser/compiler.rb:155:in `compile'
/Users/justin/Backup/code/work/server/ps/ruby/puppet/lib/puppet/parser/compiler.rb:35:in `compile'
/Users/justin/Backup/code/work/server/ps/ruby/puppet/lib/puppet/indirector/catalog/compiler.rb:305:in `compile'
/Users/justin/Backup/code/work/server/ps/ruby/puppet/lib/puppet/util/profiler/around_profiler.rb:58:in `profile'
/Users/justin/Backup/code/work/server/ps/ruby/puppet/lib/puppet/util/profiler.rb:51:in `profile'
/Users/justin/Backup/code/work/server/ps/ruby/puppet/lib/puppet/indirector/catalog/compiler.rb:303:in `compile'
/Users/justin/Backup/code/work/server/ps/ruby/puppet/lib/puppet/util.rb:237:in `benchmark'
/Users/justin/Backup/code/work/server/ps/ruby/puppet/lib/puppet/indirector/catalog/compiler.rb:301:in `compile'
/Users/justin/Backup/code/work/server/ps/ruby/puppet/lib/puppet/indirector/catalog/compiler.rb:54:in `find'
/Users/justin/Backup/code/work/server/ps/ruby/puppet/lib/puppet/indirector/indirection.rb:193:in `find'
/Users/justin/Backup/code/work/server/ps/ruby/puppet/lib/puppet/network/http/api/indirected_routes.rb:121:in `do_find'
/Users/justin/Backup/code/work/server/ps/ruby/puppet/lib/puppet/network/http/api/indirected_routes.rb:48:in `call'
/Users/justin/Backup/code/work/server/ps/ruby/puppet/lib/puppet/context.rb:65:in `override'
/Users/justin/Backup/code/work/server/ps/ruby/puppet/lib/puppet.rb:260:in `override'
org/jruby/RubyArray.java:1613:in `each'
/Users/justin/Backup/code/work/server/ps/ruby/puppet/lib/puppet/network/http/api/indirected_routes.rb:47:in `call'
/Users/justin/Backup/code/work/server/ps/ruby/puppet/lib/puppet/network/http/route.rb:82:in `process'
/Users/justin/Backup/code/work/server/ps/ruby/puppet/lib/puppet/network/http/route.rb:81:in `process'
/Users/justin/Backup/code/work/server/ps/ruby/puppet/lib/puppet/network/http/route.rb:87:in `process'
/Users/justin/Backup/code/work/server/ps/ruby/puppet/lib/puppet/network/http/route.rb:87:in `process'
/Users/justin/Backup/code/work/server/ps/ruby/puppet/lib/puppet/network/http/handler.rb:64:in `process'
/Users/justin/Backup/code/work/server/ps/ruby/puppet/lib/puppet/util/profiler/around_profiler.rb:58:in `profile'
/Users/justin/Backup/code/work/server/ps/ruby/puppet/lib/puppet/util/profiler.rb:51:in `profile'
Puppet$$Server$$Master_273493015.gen:13:in `handleRequest'
request_handler_core.clj:273:in `invoke'
jruby_request.clj:48:in `invoke'
jruby_request.clj:33:in `invoke'
request_handler_service.clj:47:in `handle_request'
request_handler.clj:3:in `invoke'
request_handler.clj:3:in `invoke'
core.clj:2515:in `invoke'
ring_middleware.clj:284:in `invoke'
core.clj:168:in `invoke'
core.clj:211:in `invoke'
core.clj:45:in `invoke'
core.clj:357:in `invoke'
core.clj:51:in `invoke'
ringutils.clj:83:in `invoke'
master_core.clj:736:in `invoke'
ring.clj:20:in `invoke'
ring.clj:12:in `invoke'
comidi.clj:249:in `invoke'
http.clj:152:in `invoke'
http.clj:152:in `invoke'
http.clj:148:in `invoke'
comidi.clj:336:in `invoke'
jetty9_core.clj:426:in `invoke'
normalized_uri_helpers.clj:74:in `invoke'
2018-01-11 11:13:36,944 WARN  [qtp1685379121-27] [puppetserver] Facter Error loading fact /Users/justin/Backup/code/work/server/ps/ruby/facter/lib/facter/macosx.rb: no such file to load -- cfpropertylist
....
```

Reply:
Looks like `puppetlabs.general-puppet.general-puppet-int-test` and `puppetlabs.services.jruby.puppet-environments-int-test` are the main culprits.

Reply:
Note, I waited to respond so I could finish up https://github.com/puppetlabs/puppetserver/pull/1599. This round of testing above is using the updated code in cbe56d6.

Reply:
Ah i see what happened. The fact lookup failures don't cause test failures, but they do cause lots and lots of stacktraces.